src/Biodiversity/Traitfuns.jl

```diff
@@ -79,3 +79,15 @@ Function to extract the trait relationship of all species in the ecosystem.
 function getrelationship(rel::R, field::Symbol) where R <: AbstractTraitRelationship
   return getfield(rel, field)
 end
+
+
+function gettraits(eco::A) where A <: AbstractEcosystem
+    return _gettraits(eco.spplist)
+end
+
+function _gettraits(sppl::SpeciesList)
+    return sppl.species.traits
+end
+function _gettraits(sppl::SpeciesList{A, B, C}) where {A <: HumanTypes, B <: VirusTypes, C}
+    return sppl.pathogens.traits
+end
```

Comment:
Yes this is a bit tricky currently - for the COVID model (i.e. `HumanTypes` + `VirusTypes`), I need to access virus traits because it is the virus that is directly impacted by the environment. But for the biodiversity (and presumably other models), I need the species themselves to respond.

Reply:
I think this may be the only thing I can see that needs to be resolved before we merge. It's not really a good solution, and there's a danger we'll be stuck with it for a while if we accept it now.